src/Config.ts

```diff
@@ -149,16 +202,85 @@ export class Config {
 
   public mosaic: MosaicConfig;
 
+  private _originWeb3?: any;
+
+  private _auxiliaryWeb3?: any;
+
   /**
    * Constructor.
-   * @param mosaicConfigPath Mosaic config path.
-   * @param chainId Auxiliary chain id.
+   * @param mosaicConfig Mosaic config object.
+   * @param facilitatorConfig Facilitator config object.
    */
   public constructor(
-    mosaicConfigPath: string,
-    chainId: string,
+    mosaicConfig: MosaicConfig,
+    facilitatorConfig: FacilitatorConfig,
   ) {
-    this.mosaic = MosaicConfig.fromPath(mosaicConfigPath);
-    this.facilitator = FacilitatorConfig.from(chainId);
+    this.facilitator = facilitatorConfig;
+    this.mosaic = mosaicConfig;
+  }
+
+  /**
+   * Returns web3 provider for origin chain.
+   */
+  public get originWeb3(): any {
+    if (this._originWeb3) {
+      return this._originWeb3;
+    }
+    const originChain = this.facilitator.chains[this.facilitator.originChainId];
+    if (!originChain.password) {
+      throw new WorkerPasswordNotFoundException(`password not found for ${originChain.worker}`);
+    }
+    const account = new Account(originChain.worker, this.facilitator.encryptedAccounts[originChain.rpc]);
+    this._originWeb3 = new Web3(originChain.rpc);
+    account.unlock(this._originWeb3, originChain.password);
+    return this._originWeb3;
+  }
+
+  /**
+   * Returns web3 provider for auxiliary chain.
+   */
+  public get auxiliaryWeb3(): any {
+    if (this._auxiliaryWeb3) {
+      return this._auxiliaryWeb3;
+    }
+    const auxiliaryChain = this.facilitator.chains[this.facilitator.auxiliaryChainId];
+    if (!auxiliaryChain.password) {
+      throw new WorkerPasswordNotFoundException(`password not found for ${auxiliaryChain.worker}`);
+    }
+    const account = new Account(auxiliaryChain.worker, this.facilitator.encryptedAccounts[auxiliaryChain.rpc]);
```

Comment:
There are code duplication in originWeb3 and auxiliaryWeb3 methods. We can refactor the common pieces to a private method.